lib/dns/records.js

```diff
@@ -1043,6 +1067,32 @@ class Addr extends Struct {
       return this;
     }
 
+    if (util.equal(this.currency, 'cosmoshub.')) {
+      const {hrp, version, hash} = bech32.decode(this.address);
+
+      assert(hash.length >= 1);
+      assert(hash.length <= 128);
+
+      let flag = 0x00;
```

Comment:
Are there expected to be more than 16 hrps? If not, I think we can fit this encoding into a nibble and save a byte?

Reply:
Scratch this; my mistake. While we can save 4 bits, @tynes reminded me we still need to encode a between 1 and 128 for the address length. We'll still need 2 bytes.

Reply:
https://github.com/handshake-org/hsd/pull/284#issuecomment-545093156